src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -178,33 +168,17 @@ ucc_status_t ucc_tl_mlx5_team_get_scores(ucc_base_team_t *  tl_team,
 
     team_info.alg_fn              = NULL;
     team_info.default_score       = UCC_TL_MLX5_DEFAULT_SCORE;
-    team_info.init                = NULL;
+    team_info.init                = ucc_tl_mlx5_coll_init;
     team_info.num_mem_types       = 2;
     team_info.supported_mem_types = mt;
     team_info.supported_colls     = UCC_TL_MLX5_SUPPORTED_COLLS;
     team_info.size                = UCC_TL_TEAM_SIZE(team);
 
-    status = ucc_coll_score_alloc(&score);
-    if (UCC_OK != status) {
-        tl_error(lib, "failed to alloc score_t");
-        return status;
-    }
-
-    status = ucc_coll_score_add_range(
-        score, UCC_COLL_TYPE_ALLTOALL, UCC_MEMORY_TYPE_HOST, 0,
-        MAX_MSG_SIZE * UCC_TL_TEAM_SIZE(team), UCC_TL_MLX5_DEFAULT_SCORE,
-        ucc_tl_mlx5_coll_init, tl_team);
-    if (UCC_OK != status) {
-        tl_error(lib, "failed to add range to score_t");
-        return status;
-    }
-
-    status = ucc_coll_score_add_range(
-        score, UCC_COLL_TYPE_ALLTOALL, UCC_MEMORY_TYPE_CUDA, 0,
-        MAX_MSG_SIZE * UCC_TL_TEAM_SIZE(team), UCC_TL_MLX5_DEFAULT_SCORE,
-        ucc_tl_mlx5_coll_init, tl_team);
+    status = ucc_coll_score_build_default(
+        tl_team, UCC_TL_MLX5_DEFAULT_SCORE, ucc_tl_mlx5_coll_init,
+        UCC_TL_MLX5_SUPPORTED_COLLS, mt, 2, &score);
```

Comment:
did you decide to keep UCC_MEMORY_TYPE_CUDA?

Reply:
yes, see this [comment](https://github.com/openucx/ucc/pull/815#discussion_r1297479919)